lib/ace/edit_session.js

```diff
@@ -1116,6 +1118,7 @@ var EditSession = function(text, mode) {
      *
      **/
     this.insert = function(position, text) {
+        this.$bidiHandler.markAsDirty();
```

Comment:
Frankly speaking I do not understand this proposal. 
Line 250 of edit-session.js falls with 'onChangeFold' method that has no bearing on our subject.
The intended changes of mine (invocation of 'markAsDirty') serve to rebuilt Bidi info pertinent to line being edited, in order to be able to correctly calculate caret position.